add example to redshift grants section (#5156)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/grants.md

<div warehouse="Redshift">

> For Redshift, you need to explicitly grant roles and groups access to your models, whether to single users, groups, or roles, and you can revoke access as well.

Reviewer: Would this be better suited here with a link from the note https://docs.getdbt.com/reference/resource-configs/redshift-configs or maybe at least also there?

<div warehouse="Redshift">

For Redshift, you need to distinguish between users, roles, and groups. It is important to explicitly grant roles and groups access to your models. You can revoke access as well.

**For users:**
When granting access to individual users, use the `+grants` syntax in your configuration to add or modify permissions for these users without replacing existing configurations. In the following example, you only need to list the users:

```yaml
models:
  my_schema:
    schema: my_schema
    description: "My schema"
    +grants:
      select: ["gspider"]  # example of user access
```

**For roles and groups:**

> For roles and groups, the `+grants` syntax is not supported. You need to provide the complete list of roles or groups that require access each time you make a change. Additionally, you must prefix each role or group with 'role' or 'group':

Reviewer 2: I don't think that the `+` is what is not supported here, rather, what isn't supported is the [inheritance of grants](https://docs.getdbt.com/reference/resource-configs/grants#grant-config-inheritance) 

```suggestion
Inheritance of grants are not supported for groups and roles in Redshift. Accordingly, for every model, you need to provide the complete list of roles or groups that require access each time you make a change to a model.

Additionally, must prefix each role or group with 'role' or 'group':
```

> ```

Author: ```suggestion
```yaml
```

<div warehouse="Redshift">

For Redshift, you need to distinguish between users, roles, and groups. It is important to explicitly grant roles and groups access to your models. You can revoke access as well.

**For users:**
When granting access to individual users, use the `+grants` syntax in your configuration to add or modify permissions for these users without replacing existing configurations. In the following example, you only need to list the users:

```yaml
models:
  my_schema:
    schema: my_schema
    description: "My schema"
    +grants:
      select: ["gspider"]  # example of user access
```

**For roles and groups:**
For roles and groups, the `+grants` syntax is not supported. You need to provide the complete list of roles or groups that require access each time you make a change. Additionally, you must prefix each role or group with 'role' or 'group':

```

> yaml

Author: ```suggestion
```

<div warehouse="Redshift">

For Redshift, you need to distinguish between users, roles, and groups. It is important to explicitly grant roles and groups access to your models. You can revoke access as well.

**For users:**
When granting access to individual users, use the `+grants` syntax in your configuration to add or modify permissions for these users without replacing existing configurations. In the following example, you only need to list the users:

```yaml
models:
  my_schema:
    schema: my_schema
    description: "My schema"
    +grants:

>       select: ["gspider"]  # example of user access

Author: 👻 🕷️ 🕸️

```

**For roles and groups:**
Inheritance of grants are not supported for groups and roles in Redshift. Accordingly, for every model, you need to provide the complete list of roles or groups that require access each time you make a change to a model.

> Additionally, must prefix each role or group with 'role' or 'group':

Author: ```suggestion
Additionally, you must prefix each role or group with 'role' or 'group':
```

```yaml
models:
  my_schema:
    schema: my_schema
    description: "Schema for reporting"
    grants:
      select: ["role my_role", "group my_group"]
```

Some things to note:

- Ensure that any roles or groups mentioned must already exist in Redshift.

> - Use the keyword `+grants` to add to or modify users for existing grants (rather than replace), however this syntax does not apply to roles and groups.

Reviewer 2: I still don't think that the `+grants` does anything different than `grants`. The important caveat is really my edit. 

```suggestion
- Adding or modifying users for existing grants (rather than replace), is not supported for roles and groups.
```

Reviewer 2: edit -- I'm totally wrong here after reading the Notes section at the end of the 
[grants: Grant config inheritance](https://docs.getdbt.com/reference/resource-configs/grants#grant-config-inheritance)

Now, I think this is the right language.

```suggestion
- The merge strategy of grants (via the `+` prefix in front of `+grants`) is not supported for roles and groups.
```
